dynamiqs/integrators/apis/mesolve.py

```diff
@@ -294,6 +300,8 @@ def _mesolve(
         Rouchon1: mesolve_rouchon1_integrator_constructor,
         Rouchon2: mesolve_rouchon2_integrator_constructor,
         Rouchon3: mesolve_rouchon3_integrator_constructor,
+        AdaptiveRouchon12: mesolve_adaptive_rouchon12_integrator_constructor,
+        AdaptiveRouchon23: mesolve_adaptive_rouchon23_integrator_constructor,
```

Comment:
I'm not a big fan of this API: if we were to implement a non-adaptive `Dopri5` (which Diffrax supports), we wouldn't rename the current `Dopri5` to `AdaptiveDopri5`. It's also quite heavy in the doc.

In other words, `AdaptiveRouchon12` is just a `Rouchon2` in an adaptive time-stepping setting. Therefore, I would be in favor of merging both under the name `Rouchon2`, and either:

- have an additional option `adaptive_stepping: bool` (or similar naming)
- make `dt: float | None = None` which activates adaptive stepping if `dt` is not specified

Reply:
Ok, it'd be great to fix this before the release you were planning then. Otherwise this will stay as is for weeks/months before a new release that will break this API.